autogpt_platform/autogpt_libs/autogpt_libs/supabase_integration_credentials_store/store.py

```diff
@@ -143,3 +153,15 @@ def _get_user_metadata(self, user_id: str) -> UserMetadataRaw:
         if not response.user:
             raise ValueError(f"User with ID {user_id} not found")
         return cast(UserMetadataRaw, response.user.user_metadata)
+
+    @contextmanager
+    def locked(self, key: str | tuple):
+        if not isinstance(key, tuple):
+            key = (key,)
+        key = (self.supabase.supabase_url, *key)
+
+        self.locks.lock(key)
```

Comment:
Isn't that the point of locking resources? It *should* clash if it's simultaneously in use elsewhere.
- The atomic unit of data here is the entire user metadata object (because it needs to be read/written in its entirety), so that dictates the lock scope: `with self.locked(user_id)`
- `locks` is shared across all `SupabaseIntegrationCredentialsStore` instances in a process
- I'm using `supabase_url` as a key prefix to account for the edge case of multiple connections to different Supabase instances

Reply:
didn't see the url prefix. That's what I meant, but since we will move this to global redis, have static prefix like "oauth-creds-:.."` or something would be usefule

Reply:
done